src/Misc.pm

```diff
@@ -5148,6 +5149,30 @@ sub completeNpcSell {
 	}
 }
 
+sub checkItemBuyNeed{
+	my $i = shift;
+	my $zeny = shift;
+							#skip block if:
+	return 0 if (!$config{"buyAuto_$i"} 			# this block is non existant
+			|| !$config{"buyAuto_$i"."_npc"}	# npc is not set
+			|| $config{"buyAuto_${i}_disabled"}	# block is not disabled
+		    );
+
+	my $amount = $char->inventory->sumByName($config{"buyAuto_$i"});# getting items amount
+
+										# item needded if:
+	return 1 if ($config{"buyAuto_$i"."_minAmount"} ne ""				# minAmount is set
+				 && $config{"buyAuto_$i"."_maxAmount"} ne ""		# maxamount is set
+				 && (checkSelfCondition("buyAuto_$i"))			# selfConditions are met
+				 && $amount <= $config{"buyAuto_$i"."_minAmount"}	# we got less than minAmount
+				 && $amount < $config{"buyAuto_$i"."_maxAmount"}	# we got less than maxAmount
+				 && ($config{"buyAuto_$i"."_price"}			# the price is set...
```

Comment:
i've read the source code, and if i understand well, you can still buy even without set price of item.
although is there on config.txt, price is not documented, and i've seen people who not use it

edit:
just tested it, it works normally without price set

since it has not defined price, it assumes it can buy 30000 on old code
the correct is
since price is not defined, assumes you can buy the maxAmout specified in buyAuto_$i_maxAmount

other thing, i just tested your PR, and when price is not set, it stills queues buyAuto, go to npc, start buy sequence, but it doesn't buy anything, and never ends buyAuto